<?xml version="1.0" encoding="utf-8"?>
<launch>
    <arg name="sim" default="true" />

	<include file="$(find optoforce_sensor)/launch/onrobot_hex_sensor.launch"/>

    <!--param name="robot_description" command="$(find xacro)/xacro.py '$(find kuka_kr10_support)/urdf/kr10r1100sixx.xacro'"/-->

<param name="robot_description" command="$(find xacro)/xacro.py '$(find fdcc)/urdf/kuka_proxxon_drilling.urdf'"/>

    <rosparam file="$(find kuka_rsi_hw_interface)/test/test_params_sim.yaml" command="load" if="$(arg sim)"/>
    <rosparam file="$(find kuka_rsi_hw_interface)/test/test_params.yaml" command="load" unless="$(arg sim)"/>

    <!-- Start node without FT sensor -->
    <node name="kuka_hardware_interface" pkg="kuka_rsi_hw_interface"
      type="kuka_hardware_interface_node" respawn="false"
      output="screen"
      required="true"/>

    <!-- Load joint controller configurations from YAML file to parameter server -->
    <rosparam file="$(find kuka_rsi_hw_interface)/config/hardware_controllers.yaml" command="load"/>
    <!-- Load standard kuka controller joint names from YAML file to parameter server -->
    <rosparam file="$(find kuka_rsi_hw_interface)/config/controller_joint_names.yaml" command="load"/>

    <!-- Load controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"
        args="position_trajectory_controller joint_state_controller --shutdown-timeout 1"/>
    <!-- Load robot state publisher -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />


    <!-- Load RSI simulation node -->
    <node name='kuka_rsi_simulator' pkg='kuka_rsi_simulator' type="kuka_rsi_simulator" args="127.0.0.1 49152" if="$(arg sim)" />

</launch>
